pkg/collector/corechecks/containers/docker/check.go

```diff
@@ -211,7 +211,12 @@ func (d *DockerCheck) runDockerCustom(sender aggregator.Sender, du docker.Client
 		if len(rawContainer.Names) > 0 {
 			containerName = rawContainer.Names[0]
 		}
-		isContainerExcluded := d.containerFilter.IsExcluded(containerName, resolvedImageName, rawContainer.Labels[kubernetes.CriContainerNamespaceLabel])
+		var annotations map[string]string
+		if pod, err := workloadmeta.GetGlobalStore().GetKubernetesPodForContainer(rawContainer.ID); err == nil {
+			annotations = pod.Annotations
```

Comment:
We check annotations where we didn't before. Does this mean that there will be more things that can be excluded with the new code?

Reply:
Indeed, we can now exclude pods or containers by adding pod annotations.
This feature was added in https://github.com/DataDog/datadog-agent/pull/16489 but did not filter out corecheck metrics.

Reply:
If a customer does not use the annotation then the behavior won't change. I don't think it would be useful to add a releasenote as the current PR is just a fix of a feature that is still not released.
There is already a releasenote for this feature: https://github.com/DataDog/datadog-agent/blob/main/releasenotes/notes/ad-exclude-annotations-0cce55729c053b9d.yaml